### DISCUSSION TITLE
Way to get queryKey in the result of useQuery

### DISCUSSION THREAD

**Discussion by rongdi1993** [TanStack/query#1878] [Q&A] [answered]
Is there a way to get `queryKey` in the result of `useQuery`? 

```
const result = useQuery(
['query key'], 
(queryKey) => {
 // I know I can use queryKey here
})
```
We have use case to map `queryKey` with the `result.data` above. I'm able to find a way that use `$response.request.params` to get the `queryKey`. But we don't have the type to know `$response` in the Typescript. The type of `result` is `QueryObserverResult`, which only includes the `data`. 
- result.data 👍 
- result.$response ❌ 

That's really painful since TypeScript will throw the error on the type definition, and we have to enforce its type to be `any`. So 
- Could we add `queryKey` in the `result` or `result.data`?
- Or could we add `$response` into the type `QueryObserverResult`?

Any idea would help. Thanks!

**Comment by TkDodo** [comment, marked as answer]
hm, I don't really understand the use-case, but what you can do is:

```
const result = useQuery(
['query key'], 
async (queryKey) => {
 const response = await (...)
 return { ...response.data, queryKey }
})
```

and then data will just have the queryKey ...
